drivers/SmartThings/zigbee-switch/src/aqara/init.lua

```diff
@@ -2,63 +2,95 @@ local capabilities = require "st.capabilities"
+local maxPower = capabilities["stse.maxPower"]
+local restorePowerState = capabilities["stse.restorePowerState"]
+local electricSwitchType = capabilities["stse.electricSwitchType"]
+local changeToWirelessSwitch = capabilities["stse.changeToWirelessSwitch"]
 
-local MFG_CODE = 0x115F
 local PRIVATE_CLUSTER_ID = 0xFCC0
 local PRIVATE_ATTRIBUTE_ID = 0x0009
+local MFG_CODE = 0x115F
+local WIRELESS_SWITCH_CLUSTER_ID = 0x0012
+local WIRELESS_SWITCH_ATTRIBUTE_ID = 0x0055
+local RESTORE_POWER_STATE_ATTRIBUTE_ID = 0x0201
+local CHANGE_TO_WIRELESS_SWITCH_ATTRIBUTE_ID = 0x0200
+local MAX_POWER_ATTRIBUTE_ID = 0x020B
+local ELECTRIC_SWITCH_TYPE_ATTRIBUTE_ID = 0x000A
 
-local PREF_CLUSTER_ID = 0xFCC0
-local PREF_MAX_POWER_ATTR_ID = 0x020B
-local PREF_RESTORE_STATE_ATTR_ID = 0x0201
-
-local ENDPOINT_POWER_METER = 0x15
-local ENDPOINT_ENERGY_METER = 0x1F
+local POWER_METER_ENDPOINT = 0x15
+local ENERGY_METER_ENDPOINT = 0x1F
 
-local APPLICATION_VERSION = "application_version"
 local LAST_REPORT_TIME = "LAST_REPORT_TIME"
+local PRIVATE_MODE = "PRIVATE_MODE"
 
 local FINGERPRINTS = {
-  { mfr = "LUMI", model = "lumi.plug.maeu01" }
+  { mfr = "LUMI", model = "lumi.plug.maeu01" },
+  { mfr = "LUMI", model = "lumi.switch.n0agl1" },
+  { mfr = "LUMI", model = "lumi.switch.n1acn1" },
+  { mfr = "LUMI", model = "lumi.switch.n2acn1" },
+  { mfr = "LUMI", model = "lumi.switch.n3acn1" },
 }
 
--- Range from 100 to 2300 (100w ~ 2300w)
--- Data type conversion table
-local max_power_data_type_table = {
-  SinglePrecisionFloat(0, 6, 0.5625),
-  SinglePrecisionFloat(0, 7, 0.5625),
-  SinglePrecisionFloat(0, 8, 0.171875),
-  SinglePrecisionFloat(0, 8, 0.5625),
-  SinglePrecisionFloat(0, 8, 0.953125),
-  SinglePrecisionFloat(0, 9, 0.171875),
-  SinglePrecisionFloat(0, 9, 0.3671875),
-  SinglePrecisionFloat(0, 9, 0.5625),
-  SinglePrecisionFloat(0, 9, 0.7578125),
-  SinglePrecisionFloat(0, 9, 0.953125),
-
-  SinglePrecisionFloat(0, 10, 0.07421875),
-  SinglePrecisionFloat(0, 10, 0.171875),
-  SinglePrecisionFloat(0, 10, 0.26953125),
-  SinglePrecisionFloat(0, 10, 0.3671875),
-  SinglePrecisionFloat(0, 10, 0.46484375),
-  SinglePrecisionFloat(0, 10, 0.5625),
-  SinglePrecisionFloat(0, 10, 0.66015625),
-  SinglePrecisionFloat(0, 10, 0.7578125),
-  SinglePrecisionFloat(0, 10, 0.85546875),
-  SinglePrecisionFloat(0, 10, 0.953125),
-
-  SinglePrecisionFloat(0, 11, 0.025390625),
-  SinglePrecisionFloat(0, 11, 0.07421875),
-  SinglePrecisionFloat(0, 11, 0.123046875)
+local preference_map = {
+  [restorePowerState.ID] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = RESTORE_POWER_STATE_ATTRIBUTE_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Boolean,
+  },
+  [changeToWirelessSwitch.ID] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = CHANGE_TO_WIRELESS_SWITCH_ATTRIBUTE_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Uint8,
+    value_map = { [true] = 0x00,[false] = 0x01 },
+  },
+  [maxPower.ID] = {
```

Comment:
I don't see this as a preference or capability in any of the profiles

Reply:
maxPower is used in the smart plug setting and is described in the "switch-power-energy-consumption-report-aqara.yml" profile.

Reply:
that's not included in this pull request

Reply:
Ah. I see.